handsontable/src/plugins/filters/ui/multipleSelect.js

```diff
@@ -322,6 +322,56 @@ export class MultipleSelectUI extends BaseUI {
     super.destroy();
   }
 
+  /**
+   * Returns the display values for all the cells at the given `column` visual index.
+   * If no display values are defined, the `source data` values will be returned instead.
+   *
+   * If the provided column index is not a valid integer, the method will return `null`.
+   *
+   * @param {number} column From visual column index.
+   * @returns {Array[]|null} Array containing the display values of all the cells in the column.
+   */
+  getDisplayValuesAtCol(column) {
+    if (!Number.isInteger(column)) {
+      return null;
+    }
+
+    const sourceRowCount = this.hot.countSourceRows();
+    const columnData = [];
+    let returnValue;
+
+    for (let row = 0; row < sourceRowCount; row += 1) {
+      const cellMeta = this.hot.getCellMeta(row, column);
+      const {
+        prop,
+        displayValue
+      } = cellMeta;
+      const cellSourceData = this.hot.getSourceDataAtCell(row, column);
+
+      if (displayValue) {
```

Comment:
When the `displayValue` value is `0` (`false` and maybe others), then it's not rendered. I'd try to render all values except `undefined` and only then return source data.